### DISCUSSION TITLE
Dynamic import returns undefined when using variable query string

### DISCUSSION THREAD

**Discussion by mattpodolak** [webpack/webpack#15117] [Q&A]
# Bug report

**What is the current behavior?**
Dynamic import returns undefined when using variable query string, but when using normal string it works as expected

**If the current behavior is a bug, please provide the steps to reproduce.**
bring your own svg (or use this https://freesvg.org/nicubunu-potato) and use the following code:

```
import React, { useState, useEffect, useRef } from 'react';

function useDynamicSVGImport(name) {
  const ImportedIconRef = useRef();
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState();

  useEffect(() => {
    setLoading(true);
    const importIcon = async () => {
      try {
        ImportedIconRef.current = (
          await import( '../icons/' + name + '.svg')
        ).ReactComponent;
      } catch (err) {
        setError(err);
        throw err;
      } finally {
        setLoading(false);
      }
    };
    importIcon();
  }, [name]);

  return { error, loading, SvgIcon: ImportedIconRef.current };
}

const Icon = ({ name, ...rest }) => {
  const { loading, SvgIcon } = useDynamicSVGImport(name);

  if (loading) {
    return null;
  }

  if (SvgIcon) {
    return <SvgIcon {...rest} />;
  }
  return null;
};


<Icon name="icon" />
```

replacing import with `await import( '../icons/icon.svg')` works

**What is the expected behavior?**
it should return the svg

**Other relevant information:**
webpack version:  5.64.4 (using CRA)
Node.js version: 14.5.0
Operating System: Windows
Additional tools: react scripts 5.0.0

**Comment by alexander-akait** [comment]
Please create fully reproducible test repo, I can't reproduce locally. What is `ReactComponent` in your context? svgr? How you setup svgr?

**Reply by mattpodolak** [reply]
Yes its svgr, repo is setup using create react app (CRA) so svgr is using the default config from CRA

**Reply by alexander-akait** [reply]
Sorry, I don't know what is `default`... Please provide example, maybe bug in CRA, because without CRA I can't reproduce the problem

**Reply by mattpodolak** [reply]
It looks like this is how its setup:
```
{
  loader: require.resolve('@svgr/webpack'),
  options: {
    prettier: false,
    svgo: false,
    svgoConfig: {
      plugins: [{ removeViewBox: false }],
    },
    titleProp: true,
    ref: true,
  },
},
```

**Reply by alexander-akait** [reply]
Can you provide full code of the problem? Sorry, I can't reproduce...

**Reply by mattpodolak** [reply]
yeah, ill create a repo and share it here, thanks

**Reply by mattpodolak** [reply]
try running this: https://github.com/mattpodolak/webpack-svg

**Reply by alexander-akait** [reply]
Thanks, I will look at this in near future

**Reply by mattpodolak** [reply]
I looked into it a bit further, a change was made to the CRA webpack config in 5.0.0 that I believe is causing this

**Reply by alexander-akait** [reply]
svgr just not applied to dynamic import in your case, problem here:
```
issuer: {
  and: [/\.(ts|tsx|js|jsx|md|mdx)$/],
},
```

https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/config/webpack.config.js#L408

If you comment it, all works fine. There is an issue https://github.com/webpack/webpack/issues/9309

**Reply by mattpodolak** [reply]
awesome, thanks for the info!
